src/config-import.js

```diff
@@ -0,0 +1,231 @@
+import yauzl from 'yauzl-promise'
+import { validate, valueSchemas } from '@mapeo/schema'
+import { text, buffer } from 'node:stream/consumers'
+
+// Throw error if a zipfile contains more than 10,000 entries
+const MAX_ENTRIES = 10_000
+
+/**
+ * @typedef {Omit<yauzl.Entry, 'fileName'> & { filename: string }} Entry
+ */
+/**
+ * @typedef {{
+ *   presets: { [id: string]: unknown }
+ *   fields:  { [id: string]: unknown }
+ * }} PresetsFile
+ */
+/**
+ * @typedef {Parameters<import('./icon-api.js').IconApi['create']>[0]} IconData
+ */
+
+/**
+ * @param {string} configPath
+ */
+export async function readConfig(configPath) {
+  /** @type {Error[]} */
+  const warnings = []
+
+  const zip = await yauzl.open(configPath)
+  if (zip.entryCount > MAX_ENTRIES) {
```

Comment:
1. Do we want a maximum ZIP file size as well?
2. [yauzl notes](https://github.com/overlookmotel/yauzl-promise/blob/e52bdeda0fd1f0bf2292f13d7f89208e17607170/README.md#zipentrycountiscertain) that you can't always know the number of entries in a ZIP file upfront. It seems like the maximum is 65,535&mdash;much larger than `MAX_ENTRIES`&mdash;but I could see adding a comment warning future developers about increasing this limit. Alternatively, we could handle that case.